magento/magento2
# [PHPMD] Add phpmd rule with ignoring of unused required parameters in the plugins

### Description (*)

This PR improve phpmd `UnusedFormalParameter` rule with ignoring of required unused parameters in the plugins.

- `around` and `after` plugins has 2 required params `$subject` and `$proceed` (or `$result` for `after` plugin).
- `before` plugin has 1 required param `$subject`.

This PR override standard `UnusedFormalParameter` rule and allow don't use `@SuppressWarnings(PHPMD.UnusedFormalParameter)` in the plugins.

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Configure phpmd
2. Remove `@SuppressWarnings(PHPMD.UnusedFormalParameter)` from some plugin
3. Add `PluginInterface` implementation into the class
4. Run phpmd. Expected no `UnusedFormalParameter` errors

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37914: [PHPMD] Add phpmd rule with ignoring of unused required parameters in the plugins
